File: package/android/src/main/java/com/margelo/nitro/foundationmodels/HybridFoundationModels.kt
package com.margelo.nitro.foundationmodels

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip

@Keep
@DoNotStrip
class HybridFoundationModels: HybridFoundationModelsSpec() {

    override fun hello(name: String): String {
        return "Hello $name from FoundationModels!"
    }

    override fun add(a: Double, b: Double): Double {
        return a + b
    }
}

File: package/cpp/HybridFoundationModels.cpp
#include "HybridFoundationModels.hpp"

namespace margelo::nitro::foundationmodels {
  std::string HybridFoundationModels::hello(const std::string& name) {
    return "Hello " + name + " from FoundationModels!";
  }

  double HybridFoundationModels::add(double a, double b) {
    return a + b;
  }
} // namespace margelo::nitro::foundationmodels
